tools/integration_tests/util/client/storage_client.go

```diff
@@ -222,3 +222,53 @@ func StatObject(ctx context.Context, client *storage.Client, object string) (*st
 	}
 	return attrs, nil
 }
+
+// UploadGcsObject uploads a local file to a specified GCS bucket and object.
+// Handles gzip compression if requested.
+func UploadGcsObject(ctx context.Context, localPath, bucketName, objectName string, uploadGzipEncoded bool) error {
+	client, err := storage.NewClient(ctx)
+
+	if err != nil {
+		return fmt.Errorf("failed to create storage client: %v", err)
+	}
+	defer client.Close()
+
+	// Create a writer to upload the object.
+	obj := client.Bucket(bucketName).Object(objectName)
+	w := obj.NewWriter(ctx)
+	defer func() {
+		err := w.Close()
```

Comment:
This function absorbs any existing non-nil err value at this point, thus hiding error, so this shouldn't even bother doing anything if `err != nil` already.

Reply:
a new variable named err is being created here as we are using err:=w.close(), hence it should not absorb any existing non-nil err.